### PR TITLE
signup route 

### PR DESCRIPTION
**Pull request by yasmeenattallah**
Relates #22
## Sign up Route 
Path ``` server/controller/routes/user/signup.js ``` 
_____________
### Description:
The signUp Route to register a new user according to validation requirements and database.
___________________
###  to post /signup 
- [x] check user input data validation using yup schema.
- [x]  make a getUser querie in the database to check if the user exists or not.
- [x] overwrite text  password with hash password to be secured using bcrypt library .
- [x] add a new user successfully in the database tables by addUser querie .
_________________
### testing

### CODE REVIEW

File: server/controller/routes/user/signup.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const { checkUserByEmail, addUser } = require('../../../database/queries/user');
+const boomify = require('../../../utils/boomify');
+
+const signupController = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const { rows } = await checkUserByEmail(email);
+    console.log(rows, 'rr');
```
Comment (AdhamHaisami) [inline server/controller/routes/user/signup.js:11]: Is log sentence necessary?

File: server/controller/routes/user/signup.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    if (rows.length) {
+      throw boomify(409, 'user already exists ');
+    }
+    const hashedPassword = await bcrypt.hash(password, 10);
+    // const {rows:[result]} = await addUser({ ...req.body, password: hashedPassword });
```
Comment (AdhamHaisami) [inline server/controller/routes/user/signup.js:16]: Commented code must be deleted

**Review by mAbadsa — APPROVED** [review]

File: server/controller/routes/user/signup.js (hunk 1 above)
Comment (yasmeenattallah) [reply server/controller/routes/user/signup.js]: done

File: server/controller/routes/user/signup.js (hunk 2 above)
Comment (yasmeenattallah) [reply server/controller/routes/user/signup.js]: yes , done

**Review by mAbadsa — APPROVED** [review]

**Review by mariamisa — APPROVED** [review]

**Pull request merged**